.github/workflows/custard-run.yaml

```diff
@@ -0,0 +1,166 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: (experimental) Custard run
+
+on:
+  push:
+    branches:
+      - main
+  workflow_run:
+    workflows:
+      - Custard CI # .github/workflows/custard-ci.yaml
+    types:
+      - in_progress
+  workflow_dispatch:
+    inputs:
+      run-all:
+        description: Run all tests
+        type: boolean
+        default: false
+      paths:
+        description: Comma separated packages to test
+        type: string
+      ref:
+        description: Branch, tag, or commit SHA to run tests on
+        type: string
+        default: main
+  # schedule:
+  #   # https://crontab.guru/#0_12_*_*_0
+  #   - cron: 0 12 * * 0 # At 12:00 on Sunday
+
+jobs:
+  affected:
+    uses: GoogleCloudPlatform/cloud-samples-tools/.github/workflows/affected.yaml@v0.2.3
+    permissions:
+      statuses: write
+    with:
+      head-sha: ${{ github.event.workflow_run.head_sha || inputs.ref || github.sha }}
+      config-file: .github/config/nodejs.jsonc
+      paths: ${{ (inputs.run-all && '.') || inputs.paths || '' }}
+      check-name: (experimental) Custard CI
+      create-check-if: ${{ !!github.event.workflow_run }}
+
+  lint:
+    needs: affected
+    runs-on: ubuntu-latest
+    permissions:
+      statuses: write
+    timeout-minutes: 5
+    steps:
+      - name: Check in_progress
+        uses: GoogleCloudPlatform/cloud-samples-tools/actions/steps/create-check@v0.2.3
```

Comment:
I think we may get an immediate renovate PR to bump all these versions to 0.2.4, but that might be a good follow up test for running tests from forks?